Added a sample file for qrcode detection in python

jshiwam opened the pull request:
<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under OpenCV (BSD) License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake


```
force_builders_only=linux,docs
```

alalek commented on samples/python/qrcode.py:
```diff
@@ -0,0 +1,139 @@
+parser.add_argument('-m','--multi',help="use detect for multiple qr-codes (value True/False default value is True)",metavar="",type=str2bool,default=True)
+parser.add_argument('-o','--out',help="path to result file (default output filename is qr_code.png)",metavar="",default="qr_code.png")
+args = parser.parse_args()
+print(args)
```
"global" code is not allowed in OpenCV Python samples (see "demo.py", it loads these files as modules).
Move such code into the `main()` function.

alalek replied:
> Move such code into the main() function

Not resolved.

jshiwam replied:
I have updated the changes please check

alalek commented on samples/python/qrcode.py:
```diff
@@ -0,0 +1,139 @@
+if PY3:
+    xrange = range
+
+def str2bool(v):
```
Use `action='store_true'` in `.add_argument()` instead. Refer to other code in OpenCV repository.

alalek commented on samples/python/qrcode.py:
```diff
@@ -0,0 +1,139 @@
+def main(argv):
+    global o_f_name
+    global o_f_ext
+    if(args.out!=''):
```
`if` in Python doesn't require brackets

alalek commented on samples/python/qrcode.py:
```diff
@@ -0,0 +1,139 @@
+        for i in range(n):
+            cnt=np.array(points[i]).reshape((-1,1,2)).astype(np.int32)
+            drawQRCodeContours(result,cnt)
+            print ('QR[',i,']@',cnt.reshape(1,-1).tolist(),": ",end="")
```
Please use "Pythonic" code. See "string format" examples: https://docs.python.org/2.7/library/string.html#format-examples (you may replace `.format` with ` % ` too - see `drawFPS()` above).

jshiwam replied:
Done!

asmorkalov commented on samples/python/qrcode.py:
```diff
@@ -0,0 +1,157 @@
+        action='store_true')
+    parser.add_argument(
+        '-m',
+        '--multi',
```
I propose to run detectMulti by default and inverse the option like for detection case. The QR code images in OpenCV extra contains several QR codes per image and the sample started with default options produces garbage.

asmorkalov commented on samples/python/qrcode.py:
```diff
@@ -0,0 +1,157 @@
+        metavar="")
+    parser.add_argument(
+        '-d',
+        '--detect',
```
Sample executed with default options and just -d shows garbage as detection.
![qr_garbadge](https://user-images.githubusercontent.com/2536374/78868168-17e15280-7a4b-11ea-8314-3aa5dd16d894.png)

jshiwam replied:
I have enabled the multiple detection by default and used the -m tag to disable the multiple detection so that single qrcodes can be detected and decoded. Now running on the opencv_extra dataset with default parameters doesn't give garbage values. 
Please let me know if some other changes are to be made in this.

alalek commented:
> I have enabled the multiple detection by default and used the -m tag to disable the multiple detection so that single qrcodes can be detected and decoded. 

Please align behavior with C++ QR code sample.

Code should preserve behavior and should do the same things.
Also critical parts of code itself should look similar.
We should minimize confusions in general.

jshiwam commented:
> > I have enabled the multiple detection by default and used the -m tag to disable the multiple detection so that single qrcodes can be detected and decoded.
> 
> Please align behavior with C++ QR code sample.
> 
> Code should preserve behavior and should do the same things.
> Also critical parts of code itself should look similar.
> We should minimize confusions in general.



> > I have enabled the multiple detection by default and used the -m tag to disable the multiple detection so that single qrcodes can be detected and decoded.
> 
> Please align behavior with C++ QR code sample.
> 
> Code should preserve behavior and should do the same things.
> Also critical parts of code itself should look similar.
> We should minimize confusions in general.

Now I have removed the ambiguity in behaviour and implemented the detection from camera as well.
